rcmd/configure_test.go

```diff
@@ -31,61 +31,113 @@ func TestConfigureArgs(t *testing.T) {
 			"non-impactful system env set",
 			"",
 			[]string{"MISC_ENV=foo", "MISC2=bar"},
-			[]string{"R_LIBS_SITE=path/to/install/lib", "MISC_ENV=foo", "MISC2=bar"},
+			[]string{"MISC_ENV=foo", "MISC2=bar", "R_LIBS_SITE=path/to/install/lib"},
 		},
 		{
 			"non-impactful system env set with known package",
 			"dplyr",
 			[]string{"MISC_ENV=foo", "MISC2=bar"},
-			[]string{"DPLYR_ENV=true", "R_LIBS_SITE=path/to/install/lib", "MISC_ENV=foo", "MISC2=bar"},
+			[]string{"DPLYR_ENV=true", "MISC_ENV=foo", "MISC2=bar", "R_LIBS_SITE=path/to/install/lib"},
 		},
 		{
 			"impactful system env set on separate package",
 			"",
 			[]string{"MISC_ENV=foo", "MISC2=bar", "DPLYR_ENV=false"},
-			[]string{"R_LIBS_SITE=path/to/install/lib", "MISC_ENV=foo", "MISC2=bar", "DPLYR_ENV=false"},
+			[]string{"MISC_ENV=foo", "MISC2=bar", "DPLYR_ENV=false", "R_LIBS_SITE=path/to/install/lib"},
 		},
 		{
 			"impactful system env set with known package",
 			"dplyr",
 			[]string{"MISC_ENV=foo", "MISC2=bar", "DPLYR_ENV=false"},
-			[]string{"DPLYR_ENV=true", "R_LIBS_SITE=path/to/install/lib", "MISC_ENV=foo", "MISC2=bar"},
+			[]string{"DPLYR_ENV=true", "MISC_ENV=foo", "MISC2=bar", "R_LIBS_SITE=path/to/install/lib"},
 		},
 		{
 			"R_LIBS_SITE env set",
 			"",
 			[]string{"R_LIBS_SITE=original/path", "MISC2=bar"},
-			[]string{"R_LIBS_SITE=path/to/install/lib", "MISC2=bar"},
+			[]string{"MISC2=bar", "R_LIBS_SITE=path/to/install/lib",},
 		},
 		{
 			"R_LIBS_SITE env set with known package",
 			"dplyr",
 			[]string{"R_LIBS_SITE=original/path", "MISC2=bar"},
-			[]string{"DPLYR_ENV=true", "R_LIBS_SITE=path/to/install/lib", "MISC2=bar"},
+			[]string{"DPLYR_ENV=true", "MISC2=bar", "R_LIBS_SITE=path/to/install/lib"},
 		},
 		{
 			"R_LIBS_USER env set",
 			"",
 			[]string{"R_LIBS_USER=original/path", "MISC2=bar"},
-			[]string{"R_LIBS_SITE=path/to/install/lib", "MISC2=bar"},
+			[]string{"MISC2=bar", "R_LIBS_SITE=path/to/install/lib"},
 		},
 		{
 			"R_LIBS_USER env set with known package",
 			"dplyr",
 			[]string{"R_LIBS_USER=original/path", "MISC2=bar"},
-			[]string{"DPLYR_ENV=true", "R_LIBS_SITE=path/to/install/lib", "MISC2=bar"},
+			[]string{"DPLYR_ENV=true", "MISC2=bar", "R_LIBS_SITE=path/to/install/lib"},
 		},
 		{
 			"R_LIBS_SITE and R_LIBS_USER env set",
 			"",
 			[]string{"R_LIBS_USER=original/path", "R_LIBS_SITE=original/site/path", "MISC2=bar"},
-			[]string{"R_LIBS_SITE=path/to/install/lib", "MISC2=bar"},
+			[]string{"MISC2=bar", "R_LIBS_SITE=path/to/install/lib"},
```

Comment:
@dpastoor After merging in the unit test fixes from the other branch, I'm taking another glance at these and realizing that I don't fully understand the "business logic" here, and that makes me worried that these tests are misconfigured. 

Specifically, it looks like `R_LIBS_USER` is dropped or replaced with `R_LIBS_SITE` in all cases. According to this comment:
```
			// we don't want to track the order of these anyway since they should take priority in the end
			// R_LIBS_USER takes precedence over R_LIBS_SITE
			// so will cause the loading characteristics to
			// not be representative of the hierarchy specified
			// in Library/Libpaths in the pkgr configuration
			// we only want R_LIBS_SITE set to control all relevant library paths for the user to
```

`R_LIBS_USER` is supposed to get precedence, which I would assume means we want it to appear later in the final environment variable list. Our tests do not currently reflect this behavior.

I'm going to proceed writing tests as if the current tests are accurate, but please let me know if I need to make updates.